src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,432 @@
+      if (intSum2 == intSum) {
+        return true;
+
+      // If statement that detects when the count that goes to the last element of the array is greater than the count that goes to the beginning of the array. If this ever happens, it is impossible for the array to be balanced.
+      } else if (intSum2 > intSum) {
+        return false;
+      }
+
+      // After each divider loop, resets the values of both sums. Therefore, the sums can be re-added.
+      intSum = 0;
+      intSum2 = 0;
+    }
+
+    // Prevents an error in the program.
+    return false;
+  }
+
+/* 
+* Given an integer n, write a method diagonal(int n) that outputs to a text file diagonalOut.txt, a two-dimensional array of size (n×n) populated as follows, with a comma between each number:
+* @param n = User inputted value that will affect the size of the array
+* @return a changed text file. No variable will be returned to the main file.
+* @author E.Lai
+*/
+  // Array 6 - Two Dimensional
+  public static void diagonal(int n) throws IOException {
+
+    // Creating PrintWriter meathod which allows the program to write in a text file
+    PrintWriter Writing = new PrintWriter(new FileWriter("src/gr11review/part2/diagonalOut.txt"));
+
+    // Variables
+    int intLoop;
+    int intLoop2;
+
+    // Creating 2D Array
+    int[][] intArray = new int[n][n];
+
+    // 2 For loops to select every cell of the array. First for loop to count every square in a row. The second for loop to count every column underneath the squares.
+    for (intLoop = 0; intLoop < n; intLoop++) {
+      for (intLoop2 = 0; intLoop2 < n; intLoop2++) {
+
+        // Fills up the whole array with 0s
+        intArray[intLoop][intLoop2] = 0;
+
+        // If statement that checks if the cell is on a diagonal
+        if (intLoop2 == (n - intLoop - 1)){
+          intArray[intLoop][intLoop2] = 1;
+
+        // If statement that checks if the cell is underneath the diagonal
+        } else if (intLoop2 > (n - intLoop - 1)) {
+          intArray[intLoop][intLoop2] = 2;
+        }
+        // Prints out the numbers
+        Writing.print(intArray[intLoop][intLoop2]);
+
+        // Prints out the commas
+        if (intLoop2 != n-1) {
+          Writing.print(",");
+        }
+      }
+
+      // Prints out the commas in each row
+      Writing.println(); 
+    }
+
+    // Closes Text File
+    Writing.close();
+  }
+
+  /**
+  * Given a string, looks for a pattern that has a length of 3, starts with "z", and ends with "p"
+  * Creates a new string with the middle character between "z" and "p" taken out (Method 1)  
+  * 
+  * @param str The string that is analyzed for the pattern 
+  * @return The new string
+  * @author N.Lee
+  */
+  public static String zipZap(String str){
+
+    // Declares variables
+    int intLen;
+    String strNewWord = " ";
     
-}
\ No newline at end of file
+    // Added two spaces to the end of the string which adds two extra indices
```

Comment:
@NicholasLee03 functional, but not to specification.  Strip the leading and trailing spaces when you are done.  
```
Test: Utility.zipZap("zipzap") ==> expected: <zpzp> but was: < zpzp  >
Test: Utility.zipZap("zipXzap") ==> expected: <zpXzp> but was: < zpXzp  >
Test: Utility.zipZap("zzzopXzzop") ==> expected: <zzzzpXzzp> but was: < zzzzpXzzp  >
Test: Utility.zipZap("ziiiipzap") ==> expected: <ziiiipzp> but was: < ziiiipzp  >
```